### PR TITLE
Update design token format

### PR DESCRIPTION
**Pull request by EllyLoel**
I've updated the format of the design tokens to match the W3C Design Tokens CG's format https://tr.designtokens.org/format/ so we're able to use tools like [Style Dictionary](https://amzn.github.io/style-dictionary/) to easier create more generators (like #353).

### CODE REVIEW

File: src/export/export-chakra.ts
Comment (spykr) [inline src/export/export-chakra.ts:144]: Should we add some documentation to this function to make it clear what it's doing?

Comment (spykr) [reply src/export/export-chakra.ts:144]: Also let's get rid of the `any` here. Is the `any` only necessary here because our `Tokens` type is inaccurate?

File: src/export/export-chakra.ts
Comment (spykr) [inline src/export/export-chakra.ts:145]: If the goal of reassigning here is to avoid mutating the original object then we need to spread it. I also think we should rename the variable to `newTokens` to make it more explicit. e.g.

```
const newTokens = { ...tokens };
```

File: src/export/export-chakra.ts
Comment (spykr) [inline src/export/export-chakra.ts:128]: Looks like the added code here is redundant, can we revert it?

File: src/export/export-chakra.ts
Comment (spykr) [inline src/export/export-chakra.ts:187]: The colour values of the shadows are much less readable now, do we think that's a problem worth fixing?

image

File: src/import/import-figma.ts
Comment (spykr) [inline src/import/import-figma.ts:612]: I'm curious why you moved the wrapping quotes to this stage of the process rather than at the point of rendering it to a CSS property, is this something mandated by the Style Dictionary format?

File: src/import/import-figma.ts
Comment (spykr) [inline src/import/import-figma.ts:375]: I'm curious why you moved the wrapping quotes to this stage of the process rather than at the point of rendering it to a CSS property, is this something mandated by the Style Dictionary format?

File: src/import/import-figma.ts
Comment (spykr) [inline src/import/import-figma.ts:717]: As far as I can tell it's not actually possible for `typeof variants[k] !== 'string'` to evaluate to `false`. If that's true can we update the type definition and remove this check?

File: src/import/import-figma.ts
Comment (spykr) [inline src/import/import-figma.ts:719]: Rather than using a `@ts-ignore` to disable TypeScript can we update the type definition instead?

**Review by spykr — CHANGES_REQUESTED** [review]
Amazing work @EllyLoel, such a good change 🙌 

Just left a few comments and questions on things we might want to change before merging. Please reach out to me for assistance when you need 😊

File: src/export/export-chakra.ts
Comment (EllyLoel) [reply src/export/export-chakra.ts:144]: This one is likely due to a lack of typescript knowledge on my part.
When replacing `any` with `Tokens` I get a lot of type errors that I'm not 100% sure how to fix.

File: src/export/export-chakra.ts
Comment (EllyLoel) [reply src/export/export-chakra.ts:145]: 🤦🏼‍♀️ omg of course, lol, thanks for reminding me how JS works 😂 
This likely explains my type issues in this comment https://github.com/PortableStudios/figma2theme/pull/352#discussion_r1022275855

File: src/export/export-chakra.ts
Comment (EllyLoel) [reply src/export/export-chakra.ts:187]: Yes and no, [the spec](https://tr.designtokens.org/format/#color:~:text=The%20value%20MUST%20be%20a%20string%20containing%20a%20hex%20triplet/quartet%20including%20the%20preceding%20%23%20character.%20To%20support%20other%20color%20spaces%2C%20such%20as%20HSL%2C%20translation%20tools%20SHOULD%20convert%20color%20tokens%20to%20the%20equivalent%20value%20as%20needed.) currently says that colours have to be hex but that it likely to change as there is discussions about using other colour gamuts happening https://github.com/design-tokens/community-group/issues/137.
So I think it's probably best for us to leave it like this for now and then update to a more readable syntax when the spec changes.

File: src/import/import-figma.ts
Comment (EllyLoel) [reply src/import/import-figma.ts:375]: I moved it as there aren't any situations (that I can think of) where we need the font family without the quotes around it, so this way we don't have to go in for each exporter and make sure to add the quotes.
I'm not heart bound to this though, have to change it if you've got a reason to.

File: src/import/import-figma.ts
Comment (EllyLoel) [reply src/import/import-figma.ts:719]: This is likely the same situation as https://github.com/PortableStudios/figma2theme/pull/352#discussion_r1022275855, I don't know typescript that well so I wasn't sure how to fix the error I was getting:
```
Property '$value' does not exist on type 'string | { $value: TextVariant; }'.
  Property '$value' does not exist on type 'string'.ts(2339)
```

File: src/export/export-chakra.ts
Comment (spykr) [reply src/export/export-chakra.ts]: It looks like the TypeScript errors are because our type definition for `Tokens` doesn't match the actual object 🤔 

Here's my attempt at creating a simplistic but accurate definition:

```

```

Note that `Dictionary<T>` is just a shortcut to say "an object with `string`-type keys and `T`-type values".

I'll leave it up to you to do this in a smarter way (e.g. updating the existing `Breakpoint`, `Radii`, etc. types instead of getting rid of them) and to update the rest of the code if it throws a fit about the new type definition 😊 Reach out to me if you have any issues and we can pair on it.

File: src/export/export-chakra.ts
Comment (spykr) [inline src/export/export-chakra.ts:161]: Since we updated `reshapeDesignTokens` to no longer mutate the passed argument, we'll also need this update this line to:

```
newTokens[key] = reshapeDesignTokens(newTokens[key]);
```

File: src/export/export-chakra.ts
Comment (spykr) [reply src/export/export-chakra.ts]: Thought about this more and it may be more complex than even I expected 🤔 Sit tight and we'll discuss the best approach in our 1:1 😊

**Review by spykr — APPROVED** [review]
@EllyLoel 

I've pushed up a commit to bring the TypeScript types up-to-date, including the change to the `$type` property that we discussed 🙌 

The tests are passing now and I'm _pretty sure_ I haven't broken anything but please give it a look and a test and make sure you're happy with it before we merge 😊 

The commit: https://github.com/PortableStudios/figma2theme/pull/352/commits/1bc20694722939b866e75e1f44065bdd3f814ec2

As always hit me up if you have any questions.

**Comment by EllyLoel** [comment]
Tested and works wonderfully! 🎉  Awesome work @spykr, thanks for the help.

**Pull request merged**
